[BUG] Levant doesn't correctly determine eval failures on constraints

When using constraints such as:

```
constraint {
  attribute = "${attr.kernel.name}"
  value     = "linux"
}
```

And attempting a deployment onto Nomad running in dev mode on a OSX machine; the following log line is presented to the user:

```
[ERROR] levant/deploy: task group hashi-ui failed to place 1 allocs, failed on [] and exhausted []
[ERROR] levant/deploy: evaluation c02b8915-7dba-82b7-4e05-1e8ddbc4fdcc finished with status complete but failed to place allocations
```

When inspecting the allocation directly the actual details are discovered:
```
Placement Failure
Task Group "hashi-ui":
  * Constraint "${attr.kernel.name} = linux" filtered 1 nodes
```

Levant therefore needs to be able to catch and log evaluation failures of this kind.

--- a/levant/deploy.go
+++ b/levant/deploy.go
@@ -147,20 +147,39 @@ func (c *nomadClient) evaluationInspector(evalID *string) error {
 				return nil
 			}
 
-			var class, dimension []string
-
 			for group, metrics := range evalInfo.FailedTGAllocs {
 
-				// Iterate the classes and dimensions to generate lists of each failure.
-				for c := range metrics.ClassExhausted {
-					class = append(class, c)
-				}
-				for d := range metrics.DimensionExhausted {
-					dimension = append(dimension, d)
+				// Check if any nodes have been exhausted of resources and therfore are
+				// unable to place allocs.
+				if metrics.NodesExhausted > 0 {
+					var exhausted, dimension []string
+					for e := range metrics.ClassExhausted {
+						exhausted = append(exhausted, e)
+					}
+					for d := range metrics.DimensionExhausted {
+						dimension = append(dimension, d)
+					}
+					logging.Error("levant/deploy: task group %s failed to place allocs, failed on %v and exhausted %v",
+						group, exhausted, dimension)
 				}
 
-				logging.Error("levant/deploy: task group %s failed to place %v allocs, failed on %v and exhausted %v",
-					group, metrics.CoalescedFailures+1, class, dimension)
+				// Check if any node classes were filtered causing alloc placement
+				// failures.
+				if len(metrics.ClassFiltered) > 0 {
+					for f := range metrics.ClassFiltered {
+						logging.Error("levant/deploy: task group %s failed to place %v allocs as class \"%s\" was filtered",
+							group, len(metrics.ClassFiltered), f)
+					}
+				}
+
+				// Check if any node constraints were filtered causing alloc placement
+				// failures.
+				if len(metrics.ConstraintFiltered) > 0 {
+					for cf := range metrics.ConstraintFiltered {
+						logging.Error("levant/deploy: task group %s failed to place %v allocs as constraint \"%s\" was filtered",
+							group, len(metrics.ConstraintFiltered), cf)
+					}
+				}
 			}
 
 			return fmt.Errorf("evaluation %v finished with status %s but failed to place allocations",

